src/sgx/libpoet_enclave/poet_enclave.cpp

```diff
@@ -0,0 +1,690 @@
+
+        jret = json_object_dotset_value(waitCertObject, "duration", json_value_init_array());
+        sp::ThrowIf<sp::RuntimeError>(jret != JSONSuccess, "Unable to initialize JSON array");
+
+        duration = json_object_get_array(waitCertObject, "duration");
+
+        for(int i=0; i<DURATION_LENGTH_BYTES; i++)
+            json_array_append_number( duration, gWaitCertData.waitCert.duration[i] ) ;
+
+        jret = json_object_dotset_string(waitCertObject, "previous_block_id", gWaitCertData.waitCert.previous_block_id.c_str());
+        sp::ThrowIf<sp::RuntimeError>( jret != JSONSuccess, "WaitCertificate serialization failed on PreviousBlockId.");
+
+        jret = json_object_dotset_string(waitCertObject, "block_summary", gWaitCertData.waitCert.block_summary.c_str());
+        sp::ThrowIf<sp::RuntimeError>( jret != JSONSuccess, "WaitCertificate serialization failed on BlockSummary.");
+
+        jret = json_object_dotset_string(waitCertObject, "validator_id", gWaitCertData.waitCert.validator_id.c_str());
+        sp::ThrowIf<sp::RuntimeError>( jret != JSONSuccess, "WaitCertificate serialization failed on ValidatorId.");
+
+        //serialize waitCert object
+        size_t serializedSize = json_serialization_size(waitCertValue);
+        sp::ThrowIf<sp::ValueError>( outSerializedWaitCertificateLen < serializedSize, "WaitCertificate buffer is too small");
+
+        jret = json_serialize_to_buffer( waitCertValue, outSerializedWaitCertificate, outSerializedWaitCertificateLen );
+        sp::ThrowIf<sp::RuntimeError>( jret != JSONSuccess, "WaitCertificate serialization failed.");
+
+        Intel::SgxEcc256StateHandle eccStateHandle;
+            
+        sgx_status_t ret = sgx_ecc256_open_context(&eccStateHandle);
+        sp::ThrowSgxError(ret, "Failed to create ECC256 context");
+            
+        //sign serialized wait certificate
+        ret =   sgx_ecdsa_sign(
+                reinterpret_cast<const uint8_t *>(outSerializedWaitCertificate),
+                static_cast<int32_t>(strlen(outSerializedWaitCertificate)),
+                const_cast<sgx_ec256_private_t *>(&gValidatorSignupData.privateKey),
+                outWaitCertificateSignature,
+                eccStateHandle);
+            
+        sp::ThrowSgxError(ret, "Failed to sign wait certificate");
+    } catch (sp::PoetError& e) {
+        Log(POET_LOG_ERROR, "Error in poet enclave(ecall_FinalizeWaitCertificate): %04X -- %s", e.error_code(), e.what());
+        ocall_SetErrorMessage(e.what());
+        result = e.error_code();
+    } catch (...) {
+        Log(POET_LOG_ERROR, "Unknown error in poet enclave(ecall_FinalizeWaitCertificate)");
+        result = POET_ERR_UNKNOWN;
+    }
+    return result;
+} //ecall_FinalizeWaitCertificate
+
+
+
+poet_err_t ecall_VerifyWaitCertificate(
+    const char* inSerializedWaitCertificate,
+    const sgx_ec256_signature_t* inWaitCertificateSignature,
+    const sgx_ec256_public_t* inPoetPublicKey
+    )
+{
```

Comment:
Please restore the old code where you were receiving the prev block id, the validator id and the block summary and use them to verify with the contents of the serializedwaitcertificate you are verifying here.